src/vs/workbench/browser/parts/editor/editorCommands.ts

```diff
@@ -384,20 +385,40 @@ function registerDiffEditorCommands(): void {
 		const activeTextDiffEditor = getActiveTextDiffEditor(accessor);
 
 		if (activeTextDiffEditor) {
+			const original = activeTextDiffEditor.getControl()?.getOriginalEditor();
+			const modified = activeTextDiffEditor.getControl()?.getModifiedEditor();
+
+			let source: ICodeEditor | undefined;
+			let destination: ICodeEditor | undefined;
 			switch (mode) {
 				case FocusTextDiffEditorMode.Original:
-					activeTextDiffEditor.getControl()?.getOriginalEditor().focus();
+					source = modified;
+					destination = original;
 					break;
 				case FocusTextDiffEditorMode.Modified:
-					activeTextDiffEditor.getControl()?.getModifiedEditor().focus();
+					source = original;
+					destination = modified;
 					break;
 				case FocusTextDiffEditorMode.Toggle:
-					if (activeTextDiffEditor.getControl()?.getModifiedEditor().hasWidgetFocus()) {
+					if (modified?.hasWidgetFocus()) {
 						return focusInDiffEditor(accessor, FocusTextDiffEditorMode.Original);
 					} else {
 						return focusInDiffEditor(accessor, FocusTextDiffEditorMode.Modified);
 					}
 			}
+
+			if (destination) {
+				destination.focus();
+
+				const configurationService = accessor.get(IConfigurationService);
+				const keepLine = configurationService.getValue<boolean>('diffEditor.keepLine');
+				if (keepLine) {
+					const position = source?.getPosition();
+					if (position) {
+						destination.setPosition(position);
```

Comment:
Hmm, you probably want to consider previous diffs here:

Lets say you compare doc1 with doc2.

doc1:
```
/**
Some header
*/
function myFunction() {
    // 1
}
```


doc2:
```
function myFunction() {
    // 1
}
```

When the cursor is at `1` in doc1 and you switch sides, you want the cursor to be at `1` in doc2.
But from my understanding, your code does not support that.

Reply:
Sounds good. I'll investigate and update the PR, thanks for the heads up! :)

Reply:
I found `getDiffLineInformationForModified` and `getDiffLineInformationForOriginal` that serve exactly our purpose, now it behaves as you described (jumps to `// 1` from both editors)